apps/src/maze/api.js

```diff
@@ -258,11 +258,15 @@ exports.loopHighlight = API_FUNCTION(function (id) {
  * whether or not we're a Bee level
  */
 exports.getNectar = API_FUNCTION(function (id) {
-  Maze.subtype.getNectar(id);
+  if (Maze.subtype.getNectar()) {
```

Comment:
Maybe this should be `tryGetNectar`, just so it's more clear that it *attempts* to get nectar?

Perhaps that would change back when you pull the termination value out of the subtype, so I'll leave it up to you.

Reply:
hmm, maybe I should update the comment.

It not only attempts to get nectar, but also does so if it can, via [this line](https://github.com/code-dot-org/code-dot-org/pull/20890/files#diff-9ad6dbd7c2e25c3431d533ab2b46fb63R400).

The way the maze code execution works (which is probably a bit more complicated than it needs to be) is that is makes all the api calls which as a side effect runs all the headless stuff, then resets the environment and runs the queued animation actions.

Reply:
That's the behavior I'd expect from a `tryFoo` method (i.e. actually do `foo`, or return false if it can't), but I suppose that expectation is not as universal as I had thought.

I think the comment is fine as is.